src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -573,6 +573,27 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             list.size(), type);
     }
 
+    void visit_Tuple(const AST::Tuple_t &x) {
+        Vec<ASR::expr_t*> tuple;
+        tuple.reserve(al, x.n_elts);
+        ASR::ttype_t *type = nullptr;
+        for (size_t i=0; i<x.n_elts; i++) {
+            this->visit_expr(*x.m_elts[i]);
+            ASR::expr_t *expr = ASRUtils::EXPR(tmp);
+            if (type == nullptr) {
+                type = ASRUtils::expr_type(expr);
+            } else {
+                if (!ASRUtils::check_equal_type(ASRUtils::expr_type(expr), type)) {
+                    throw SemanticError("All Tuple elements must be of the same type for now",
+                        x.base.base.loc);
+                }
+            }
```

Comment:
And then change this to:
```suggestion
            tuple_type.push_back(al, ASRUtils::expr_type(expr));
```